### PR TITLE
Integrate gutenberg-mobile release 1.103.0

### PR DESCRIPTION
**Pull request by geriux**
## Description
This PR incorporates the 1.103.0 release of gutenberg-mobile.
For more information about this release and testing instructions, please see the related Gutenberg-Mobile PR: https://github.com/wordpress-mobile/gutenberg-mobile/pull/6139

Release Submission Checklist

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: RELEASE-NOTES.txt
Comment (fluiddot) [inline RELEASE-NOTES.txt]: I understand this would only affect the Jetpack app, so I'd add the `[Jetpack-only]` tag to the entry. Additionally, I noticed there's an extra space that could be removed.

```suggestion
* [**] [Jetpack-only] Block Editor: Add basic support to view, relocate, and remove the Jetpack Paywall block. [https://github.com/wordpress-mobile/gutenberg-mobile/pull/6076]
```

File: build.gradle
Comment (fluiddot) [inline build.gradle]: I noticed that this reference is not pointing to the latest commit of the Gutenberg Mobile PR, the SHA should be `273a193d6c0fcb05d09e1722902c8d985f089422`. Nevertheless, since the commits pushed after this reference only introduce an E2E test fix and update the release notes, we could directly use the release Git tag once we merge the Gutenberg PRs.

**Review by fluiddot — COMMENTED** [review]
The PR looks good to me 🎊 ! I hold off on the approval until we update the release notes as mentioned [here](https://github.com/wordpress-mobile/WordPress-Android/pull/19117#discussion_r1312732203).

File: RELEASE-NOTES.txt
Comment (fluiddot) [reply RELEASE-NOTES.txt]: Additionally, we'd also need to solve the file conflicts in the release notes file.

**Comment by geriux** [comment]
> The PR looks good to me 🎊 ! I hold off on the approval until we update the release notes as mentioned https://github.com/wordpress-mobile/WordPress-Android/pull/19117#discussion_r1312732203.

What do you think If I push the required changes for the approval along with the release tag? To save some time. 😅

**Comment by fluiddot** [comment]
> What do you think If I push the required changes for the approval along with the release tag? To save some time. 😅

Sure, this approach makes total sense to me (especially to save time 😅).

**Review by fluiddot — APPROVED** [review]
LGTM 🎊 !

**Pull request merged**
